src/rum/instrumentation/DdRumReactNavigationTracking.tsx

```diff
@@ -62,5 +78,27 @@ export default class DdRumReactNavigationTracking {
         return this.navigationStateChangeListener;
     }
 
+    private static registerAppStateListener() {
+        this.appStateListener = (appStateStatus: AppStateStatus) => {
+
+            const currentRoute = this.registeredContainer?.getCurrentRoute();
+            const currentViewKey = currentRoute?.key;
+            const currentViewName = currentRoute?.name;
+
+            if (currentViewKey != null && currentViewName != null) {
+                if (appStateStatus === 'background') {
+                    DdRum.stopView(currentViewKey, new Date().getTime(), {});
+                } else if (appStateStatus === 'active') {
+                    // case when app goes into foreground, in that case navigation listener
+                    // won't be called
+                    DdRum.startView(currentViewKey, currentViewName, new Date().getTime(), {});
+                }
+            }
+        };
+
+        // AppState is singleton, so we should add a listener only once in the app lifetime
```

Comment:
minor: maybe instead of a comment, we can rename this function to `registerAppStateListenerIfNeeded` and do `if this.appStateListener == null` check inside. 
that way `listener` is added only once even if the function is called multiple times

Reply:
makes sense. I applied the change, but left comment there as well, since it gives a bit more clarity.